jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java

```diff
@@ -139,9 +143,23 @@ public int read(byte[] b, int off, int len) throws IOException
     {
         synchronized (_inputQ)
         {
+            long now=System.currentTimeMillis();
+            
             if (_blockingTimeoutAt>=0 && !isAsync())
-                _blockingTimeoutAt=System.currentTimeMillis()+getHttpChannelState().getHttpChannel().getHttpConfiguration().getBlockingTimeout();
+                _blockingTimeoutAt=now+getHttpChannelState().getHttpChannel().getHttpConfiguration().getBlockingTimeout();
 
+            int minRequestDataRate=_channelState.getHttpChannel().getHttpConfiguration().getMinRequestDataRate();
+            if (minRequestDataRate>0)
+            {
+                long period=now-_channelState.getHttpChannel().getRequest().getTimeStamp();
+                if (period>=1000)
+                {
+                    double data_rate = _contentArrived / (0.001*(now-_channelState.getHttpChannel().getRequest().getTimeStamp()));
```

Comment:
Use local variable `period` already calculated above.

I'm not sure this covers well all the cases. For example, a small whole content that arrives a little late would be rejected - I think we need to consider if the whole content already arrived.
Also, `100 Continue` would look like an attack for small contents, because of the delay between the request being created and the content arriving. 
I think we need a better "start" time; not really when the request object is created, but when it's first dispatched. Before dispatch we are non-blocking so no need to worry, that is why we need to count time after we dispatched to the application where we could block.

I was thinking if we need this with TLS too, but we are non-blocking there too, so no need.